src/cocoa/toga_cocoa/dialogs.py

```diff
@@ -132,4 +132,34 @@ def open_file(window, title, file_types, multiselect):
         filename_or_filenames = (paths if multiselect else
                                  panel.URL.path)
         return filename_or_filenames
-    return None
+
+
+def select_folder(window, title, multiselect):
+    """Cocoa select folder dialog implementation.
+
+    Args:
+        window: Window dialog belongs to.
+        title: Title of the dialog.
+        multiselect: Flag to allow multiple folder selection.
+    Returns:
+        (list) A list of folder paths.
+    """
+    dialog = NSOpenPanel.alloc().init()
+    dialog.title = title
+
+    dialog.canChooseFiles = False
+    dialog.canChooseDirectories = True
+    dialog.resolvesAliases = True
+    dialog.allowsMultipleSelection = multiselect
+
+    result = dialog.runModal()
+
+    # Ensure regardless of the result, return types remain the same so as to not
+    # require type checking logic in user code.
+    # Convert types from 'ObjCStrInstance' to 'str'.
+    if result == NSFileHandlingPanelOKButton:
```

Comment:
At column 18: (F405) 'NSFileHandlingPanelOKButton' may be undefined, or defined from star imports: .libs